crates/sui-core/src/epoch/randomness.rs

```diff
@@ -241,29 +364,60 @@ impl RandomnessManager {
                 committee.epoch()
             );
             // Load intermediate data.
+
+            // load from the deprecated table if the binary was changed during dkg.
+            #[allow(deprecated)]
+            rm.processed_messages
+                .extend(tables.dkg_processed_messages.safe_iter().map(|result| {
+                    let (pid, msg) = result.expect("typed_store should not fail");
+                    (pid, VersionedProcessedMessage::V0(msg))
+                }));
             rm.processed_messages.extend(
                 tables
-                    .dkg_processed_messages
+                    .dkg_processed_messages_v2
                     .safe_iter()
                     .map(|result| result.expect("typed_store should not fail")),
             );
+
             if let Some(used_messages) = tables
-                .dkg_used_messages
+                .dkg_used_messages_v2
                 .get(&SINGLETON_KEY)
                 .expect("typed_store should not fail")
             {
                 rm.used_messages
                     .set(used_messages.clone())
                     .expect("setting new OnceCell should succeed");
+            } else {
+                // load from the deprecated table if the binary was changed during dkg
+                #[allow(deprecated)]
+                if let Some(used_messages) = tables
+                    .dkg_used_messages
+                    .get(&SINGLETON_KEY)
+                    .expect("typed_store should not fail")
+                {
+                    rm.used_messages
+                        .set(VersionedUsedProcessedMessages::V0(used_messages.clone()))
+                        .expect("setting new OnceCell should succeed");
+                }
             }
+
+            // load from the deprecated table if the binary was changed during dkg.
+            #[allow(deprecated)]
+            rm.confirmations
+                .extend(tables.dkg_confirmations.safe_iter().map(|result| {
+                    let (pid, msg) = result.expect("typed_store should not fail");
+                    (pid, VersionedDkgConfimation::V0(msg))
+                }));
             rm.confirmations.extend(
                 tables
-                    .dkg_confirmations
+                    .dkg_confirmations_v2
                     .safe_iter()
                     .map(|result| result.expect("typed_store should not fail")),
             );
         }
 
+        // TODO: Should we call advance_dkg here?
```

Comment:
I don't think so? it's the responsibility of commit handler to call that. do you see a scenario where it could fail to do so?